python/GafferUI/NumericWidget.py

```diff
@@ -264,13 +264,18 @@ def __editingFinished( self, widget ) :
 
 		assert( widget is self )
 
+		reason = self.ValueChangedReason.Edit
+
 		# In __incrementIndex we temporarily pad with leading
 		# zeroes in order to achieve consistent editing. Revert
 		# back to our standard form now so we don't leave it in
 		# this state.
-		self.setText( self.__valueToString( self.getValue() ) )
+		try :
+			self.setText( self.__valueToString( self.getValue() ) )
+		except ValueError as e :
+			reason = self.ValueChangedReason.Invalid
```

Comment:
@johnhaddon Is this the intended use of `Invalid`?

Reply:
Ahh yeah, makes sense. I think as it wasn't the last one it confoozed my simple brain. Added `InvalidEdit`[here](https://github.com/GafferHQ/gaffer/pull/3974/commits/f3835d41b5df9bc871d40f3cc081f82414eb7433)